xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/delete.js

```diff
@@ -12,6 +12,12 @@ require.config({
   }
 });
 require(['jquery', 'xwiki-meta', 'tree'], function($, xm) {
+  // prevent the user from leaving the delete
+  $(window).on("beforeunload", function(e) {
+    e.preventDefault();
+    e.returnValue = "";
+  });
```

Comment:
I don't like this. ``delete.js`` could be loaded after the delete job is finished (e.g. if the delete job is quick) or before the delete job starts (and you can't be certain that the job will start). Moreover, if the question is not asked (because the delete job is run in non-interactive mode) the ``beforeunload`` listener is not removed.

Reply:
Do you have a suggestion to put it somewhere else then? I think I looked on `ExtensionBreakingQuestion.js` and it wasn't the proper location, but I can look again.

Reply:
I wouldn't put it anywhere because I don't think the leave confirmation is a solution. As @evalica commented on the JIRA issue, you don't have to leave the page to have problems. You can just start a new job on a new browser tab or have another user start another job. I would rather let the user navigate away and use the notification menu to indicate that there is a job waiting for an answer.

Reply:
I completely agree with you: for me this PR was really a quickfix that could help some users in such situation. I actually did it because I had the issue and wasn't understood what happened: for me the confirm dialog would have helped.

Now what you suggest should be done, but I don't know how much it will take and I don't think the solutions are conflicting.

Reply:
The issue is that it's not easy to show the confirmation dialog only when it's needed. If we show it all the time (even if the job didn't start yet, or the job is already finished or the job doesn't have any question, etc.) it becomes annoying.

Reply:
ok, then indeed it's not a good quickfix if it could become annoying for users. I'll close the PR.